### ISSUE TITLE
Report HDN_ExpressionIsProbablyNeverNull for null-coalescing operator with nullable value type

### ISSUE THREAD

**Issue by cston** [dotnet/roslyn#31659]
Should report a hidden diagnostic that the implicit null check in `s?.F` is unnecessary. (The code could be simplified to `s.Value.F` or `s.GetValueOrDefault().F`.)
```c#
#nullable enable
#pragma warning disable 649

struct S
{
    internal int F;
}

class Program
{
    static void Main()
    {
        S? s = new S();
        int? i = s?.F;
        _ = i.Value;
    }
}
```
See https://github.com/dotnet/roslyn/pull/31499#issuecomment-445685703.

**Comment by gafter** [comment]
At LDM today we decided to drop the HDN diagnostics.
